### PR TITLE
ST-221 Highlight merchant fields

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/components/card-number/CardNumber.ts [hunk 1]
```diff
@@ -168,16 +173,6 @@ export default class CardNumber extends FormField {
...
-  /**
-   * Checks the response from backend
-   */
-  public backendValidation() {
-    this._messageBus.subscribe(MessageBus.EVENTS.VALIDATE_CARD_NUMBER_FIELD, (data: any) => {
-      this.checkBackendValidity(data);
-      this.validation.validate(this._inputElement, this._messageElement);
-    });
-  }
-
```
Comment (pgs-gmartowski) [inline src/components/card-number/CardNumber.ts]: info: I moved all validation methods to Validation.ts class

Comment (james-simpson-ST) [reply src/components/card-number/CardNumber.ts]: Nice refactor :)

File: src/core/classes/MerchantFields.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  /**
+   * Checks if Merchant's form has inputs with data-st-name and  returns them.
+   */
+  public findAllMerchantInputs() {
+    const merchantFieldsNamesArray = [];
+    for (let i = 0; i < this._merchantInputs.length; ++i) {
+      // @ts-ignore
+      if (this._merchantInputs[i].dataset.hasOwnProperty(MerchantFields.DATA_ATTRIBUTE_NAME)) {
```
Comment (james-simpson-ST) [inline src/core/classes/MerchantFields.ts]: I tried to use dataset for parsing the merchant fields and its not supported in IE so I had to use hasAttribute/getAttribute/setAttribute so please update to use them

Comment (james-simpson-ST) [reply src/core/classes/MerchantFields.ts]: Might only be in IE10 which we haven't added to the pipeline yet :(

Comment (pgs-gmartowski) [reply src/core/classes/MerchantFields.ts]: yes, it looks like it's only partial support on IE10 :/ Ok, I'll change that.

File: src/components/card-number/CardNumber.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/components/card-number/CardNumber.ts]: :)

File: src/core/classes/MerchantFields.ts (hunk 2 above)
Comment (pgs-gmartowski) [reply src/core/classes/MerchantFields.ts]: done

**Pull request merged**